### DISCUSSION TITLE
dumi 如何在使用 pnpm 时，支持对 Swift、C#、Kotlin 等语言的语法高亮？

### DISCUSSION THREAD

**Discussion by GuoDapeng** [umijs/dumi#2179] [Q&A] [answered]
参考 [dumi 如何支持对 Swift、C#、Kotlin 等语言的语法高亮？](https://d.umijs.org/guide/faq#dumi-如何支持对-swiftckotlin-等语言的语法高亮)，使用 `yarn` 时是没有问题的。

我尝试使用 `pnpm` 时遇到了困难。希望能得到帮助。

[这个是我测试的代码](https://github.com/GuoDapeng/dumi_test/tree/prism-test)

尝试过程：

```shell
╭─felix at FelixMacBookPro in ~/Downloads/doc 24-07-28 - 15:43:01
╰─○ npx create-dumi
✔ Pick template type › Static Site
✔ Pick NPM client › pnpm
✔ Input project name …
✔ Input project description … A static site based on dumi
✔ Input project author (Name <email@example.com>) …
```

选择了 `pnpm`，项目创建完成，在 `.dumi/global.ts`位置创建文件，粘贴以下内容：
```ts
// .dumi/global.ts
import Prism from 'prism-react-renderer/prism';

(typeof global !== 'undefined' ? global : window).Prism = Prism;

require('prismjs/components/prism-kotlin');
require('prismjs/components/prism-csharp');
```
看上去如下图

<img width="1865" alt="截屏2024-07-28 16 09 02" src="https://github.com/user-attachments/assets/eabaa2f9-c643-4ce5-b810-13d3856c3968">

这样会报错

<img width="1469" alt="截屏2024-07-28 16 10 56" src="https://github.com/user-attachments/assets/c342c65f-a526-439c-9b83-eb8e7d77aca3">

我尝试安装了 `prism-react-renderer` 和 `prismjs` 包，并且修改了 `global.ts` 文件，看上去如下图

<img width="1865" alt="截屏2024-07-28 16 43 33" src="https://github.com/user-attachments/assets/9f927b93-722e-48e2-9f18-4d799b100a00">

页面能展示了，但是语法高亮没有生效

<img width="1469" alt="截屏2024-07-28 16 44 04" src="https://github.com/user-attachments/assets/bfd7f7c6-b2f5-449e-b418-9bb9c48d0317">

期待能帮帮我，自带的 `shell` 的高亮是没有问题的，但是 `kotlin` 的不行。

**Comment by GuoDapeng** [comment, marked as answer]
我憨批了……，我看了一下 `dumi` 包，发现 `"prism-react-renderer": "^1.3.5",` 这个，原来我用的版本不对，换成 1.3.5 就好了。

**Comment by fanhualei** [comment]
我也实验了一下，发现kotlin 不能正常显示

**Reply by GuoDapeng** [reply]
检查一下 `prism-react-renderer` 版本是不是 1.x.x 的，我这个正常能高亮 https://guodapeng.icu/example/common_button#jetpack-compose
